bot/cogs/help.py

```diff
@@ -0,0 +1,234 @@
+from inspect import Parameter
+import textwrap
+import typing as t
+
+import discord
+from discord.ext import commands, menus
+import discord.utils
+
+
+class HelpSource(menus.ListPageSource):
+    """The Help menu."""
+
+    def __init__(
+            self,
+            signature: t.Callable,
+            filter_commands: t.Coroutine,
+            prefix: str,
+            author: discord.User,
+            cogs: dict
```

Comment:
Can this really be a dictionary of anything? If not we should use `t.Dict[key_type, value_type]`.

Reply:
Ya really want it that way ? I don't feel like this will be really useful

Reply:
And it'll be much longer : full-length is :
`t.Dict[t.Optional[commands.Cog], t.List[commands.Command]]`

Reply:
Why is the key optional, you can't have 2 identic keys holding different values in a dictionary and `Optional` only means that it can also be `None`. It doesn't really make sense to use Optional on the key. which would bring the length down to this:
```py
t.Dict[commands.Cog, t.List[commands.Command]]
```
which is acceptable, but if it still seems long you can import `Cog` and `Command` directly instead of doing `commands.X`

Reply:
1) This is how it's written in the discord.py documentation
2) There may be a None key for commands which are not in a cog (like help)

Reply:
Alright, but we should still do this:
```py
t.Dict[t.Optional[Cog], t.List[Command]]
```

Reply:
Check the latest commits